### PR TITLE
typescript `src/tabs`, `src/typography`

### PR DESCRIPTION
**Pull request by toger5**
I dont know what we do with this:
## This is now done using `React.createElement`

```tsx

export const Link = forwardRef(
  (
    {
      as,
      children,
      className,
      color = "link",
      href,
      to,
      fontWeight,
      overflowEllipsis,
      ...rest
    },
    ref
  ) => {
    const Component = as || (to ? RouterLink : "a");
    let externalLinkProps;

    if (href) {
      externalLinkProps = {
        href,
        target: "_blank",
        rel: "noreferrer noopener",
      };
    }

    return (
      <Component
        {...externalLinkProps}
        {...rest}
        to={to}
        className={classNames(
          styles[color],
          styles[fontWeight],
          { [styles.overflowEllipsis]: overflowEllipsis },
          className
        )}
        ref={ref}
      >
        {children}
      </Component>
    );
  }
);
```
in particular with the line:
    `const Component = as || (to ? RouterLink : "a");`
    how to type that properly?

### CODE REVIEW

File: src/tabs/Tabs.tsx [hunk 1]
```diff
@@ -17,27 +17,40 @@ limitations under the License.
...
-import styles from "./Tabs.module.css";
+import { useTabListState, TabListState } from "@react-stately/tabs";
 import classNames from "classnames";
+import { AriaTabPanelProps, TabListProps } from "@react-types/tabs";
+import { Node } from "@react-types/shared";
 
-export function TabContainer(props) {
-  const state = useTabListState(props);
-  const ref = useRef();
+// import {Collection} from
```
Comment (SimonBrandner) [inline src/tabs/Tabs.tsx:25]: Why the comment?

File: src/typography/Typography.tsx [hunk 2]
```diff
@@ -14,25 +14,24 @@ See the License for the specific language governing permissions and
...
-      overflowEllipsis,
-      ...rest
-    },
-    ref
-  ) => {
+interface TypographyProps {
+  children: ReactNode;
+  className?: string;
+  overflowEllipsis?: boolean;
+  [index: string]: unknown;
```
Comment (SimonBrandner) [inline src/typography/Typography.tsx:27]: Why is this here?

File: src/typography/Typography.tsx [hunk 3]
```diff
@@ -14,25 +14,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React, { forwardRef } from "react";
+import React, { forwardRef, ReactNode } from "react";
```
Comment (SimonBrandner) [inline src/typography/Typography.tsx:17]: Have you made sure there are no uses of these components where we don't use the default value of `as`?

File: src/typography/Typography.tsx [hunk 4]
```diff
@@ -215,21 +170,21 @@ export const Link = forwardRef(
...
           className
-        )}
-        ref={ref}
-      >
-        {children}
-      </Component>
+        ),
+        ref: ref,
+      },
+      children
```
Comment (SimonBrandner) [inline src/typography/Typography.tsx:187]: Could you ask about this in one of the rooms?

File: src/tabs/Tabs.stories.tsx [hunk 5]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 */
 
 import React from "react";
+
```
Comment (robintown) [inline src/tabs/Tabs.stories.tsx:18]: The `Tabs` component below should be given a type of `FC<{}>`

File: src/tabs/Tabs.tsx [hunk 6]
```diff
@@ -17,27 +17,40 @@ limitations under the License.
...
+import styles from "./Tabs.module.css";
+
+interface TabContainerProps<T> extends TabListProps<T> {
+  className?: string;
+}
+
+export function TabContainer<T extends object>(props: TabContainerProps<T>) {
```
Comment (robintown) [inline src/tabs/Tabs.tsx:32]: This should have an explicit return type or be of type `FC`

File: src/tabs/Tabs.tsx [hunk 7]
```diff
@@ -17,27 +17,40 @@ limitations under the License.
...
 
-function Tab({ item, state }) {
+interface TabProps<T> {
+  item: Node<T>;
+  state: TabListState<T>;
+}
+
+function Tab<T>({ item, state }: TabProps<T>) {
   const { key, rendered } = item;
   const ref = useRef();
```
Comment (robintown) [inline src/tabs/Tabs.tsx:55]: missing a type parameter

File: src/typography/Typography.stories.tsx [hunk 8]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 */
 
 import React from "react";
+
```
Comment (robintown) [inline src/typography/Typography.stories.tsx:18]: `Typography` should be `FC` as well

File: src/typography/Typography.tsx [hunk 9]
```diff
@@ -184,12 +133,18 @@ export const Micro = forwardRef(
...
-export const Link = forwardRef(
+interface LinkProps extends TypographyProps {
+  to: string;
+  as: string;
+  color: string;
+  href: string;
+}
+export const Link = forwardRef<HTMLAnchorElement, LinkProps>(
   (
     {
```
Comment (robintown) [inline src/typography/Typography.tsx:149]: `externalLinkProps` below should get a type

**Review by robintown — COMMENTED** [review]
I mostly have the same comments as Šimon, but with a few extra

**Review by robintown — APPROVED** [review]
